libs/utils/utils.js

```diff
@@ -809,6 +809,7 @@ async function decoratePlaceholders(area, config) {
   if (!area) return;
   const nodes = findReplaceableNodes(area);
   if (!nodes.length) return;
+  area.dataset.placeholdersDecorated = 'true';
```

Comment:
This seems to be added before the placeholders are decorated. Someone from the outside might hook onto this thinking that placeholders are replaced once this class is added. I'd suggest we either change the name to show that placeholders are used in the area; or move the declaration after the placeholders have actually been replaced.